src/Factory/HalConfigFactory.php

```diff
@@ -9,14 +9,11 @@
 namespace Laminas\ApiTools\Hal\Factory;
 
 use Interop\Container\ContainerInterface;
+use Laminas\ServiceManager\Factory\FactoryInterface;
 
-class HalConfigFactory
+class HalConfigFactory implements FactoryInterface
 {
-    /**
-     * @param ContainerInterface $container
-     * @return array|\ArrayAccess
-     */
-    public function __invoke(ContainerInterface $container)
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
```

Comment:
Please also revert all these changes because the topic is still the support of PHP 8 and these interfaces for factories are not needed.